### PR TITLE
Organize documentation and add getting started

### PR DESCRIPTION
**Pull request by sheryjoe**
This is to address #530

### CODE REVIEW

File: Examples/Python/RunUseCase.py [hunk 1]
```diff
@@ -42,7 +42,7 @@
 parser.add_argument("--start_with_image_and_segmentation_data", help = "use images and segmentations data for preprocessing", action="store_true")
 parser.add_argument("--use_single_scale", help="Single scale or multi scale optimization", action="store_true")
 parser.add_argument("--tiny_test", help="Run as a short test", action="store_true")
-parser.add_argument("shapeworks_path", help="Path to ShapeWorks executables (default: "+binpath+")", nargs='?', type=str, default=binpath)
+parser.add_argument("--shapeworks_path", help="Path to ShapeWorks executables (default: "+binpath+")", nargs='?', type=str, default=binpath)
```
Comment (akenmorris) [inline Examples/Python/RunUseCase.py:45]: This change breaks the PythonTests that assumes the last optional argument is the shapeworks path.  This argument did not previous need "--shapeworks_path"

**Comment by jadie1** [comment]
I noticed a few things:

- The Q.png image isn't linked properly at /Documentation/Workflow/Optimize.md
- The xml referenced at /Documentation/Workflow/Analyze.md is now called analyze.xml instead of shapeworksview2.xml
- The links to groom, optimize, and analyze.md are now broken in the individual use case .md's like ellipse.md and femur.md

Other than that I tried to check all of the links and I think they look good. I can make these changes tomorrow if that's easiest. 
It would be good to merge [PR #537 ](https://github.com/SCIInstitute/ShapeWorks/pull/537) first if possible. That is the predefined cutting plane one and I updated femur.md for it

**Comment by sheryjoe** [comment]
I have fixed those. In general, please don't use explicit GitHub links, unless needed. The groom, optimize, and analyze links were broken because they were linked to the master. These links should be relative to the documentation folder.

**Comment by akenmorris** [comment]
When we merge this, can we do a 'squash and merge' so that the unused large GIF files don't enter into the history?

**Comment by sheryjoe** [comment]
Yes. We can.

**Comment by sheryjoe** [comment]
@jadie1 @AtefehKashani @akenmorris how about this double dash for shapeworks binpath. this should also be relfected in the RunUseCase documentation.

**Comment by akenmorris** [comment]
I don't know about the double dash for the path.  @cchriste , @jadie1 , @AtefehKashani , was there a reason it was an un-named optional final argument instead?

**Comment by sheryjoe** [comment]
In general, we should maintain consistency for input arguments. Double dashes are used for use case input arguments. It makes not sense to have one of those arguments without the double dashes. Let's either use the double dashes for the shapeworks bin path or otherwise remove the argument if it has no particular use/justification.

**Comment by AtefehKashani** [comment]
I would say it is better to have the shapeworks binpath as an argument, since it may be installed in different path and setting  the default path will break the script.

**Comment by cchriste** [comment]
I agree completely that it should be a double-dash argument. It was
originally added when there were far fewer args and no standalone
executable.

On a side note, single dash, single letter arguments are common shortcut
arguments (ex: —foobar/-f). They exist in many of our executables and while
I wouldn’t recommend using them for examples, I don’t believe their use
should be discouraged.

**Comment by akenmorris** [comment]
Sounds good.  I just didn't know if there was some rationale for the current behavior.

**Comment by sheryjoe** [comment]
Agreed! @jadie1 @AtefehKashani  you can do something like below. Not only for consistency across different tools/interfaces we provide, but also single letter args could be very convenient short cut for some users especially when running these scripts multiple times.

parser.add_argument('--verbose', '-v',
                    action='store_true',
                    help='verbosely list files processed')

**Comment by AtefehKashani** [comment]
what is this verbose argument pointing to? the level of verbosity in the pipeline (like optimization)?

**Comment by akenmorris** [comment]
What else do we need to merge this?

**Comment by sheryjoe** [comment]
> what is this verbose argument pointing to? the level of verbosity in the pipeline (like optimization)?

That was meant to be an example.

**Comment by akenmorris** [comment]
Are we ready to merge?

**Comment by sheryjoe** [comment]
I think so

**Comment by akenmorris** [comment]
What about the gifs?

**Comment by sheryjoe** [comment]
For now, How about dividing this video into three gifs to demo groom, optimize and analyze. Do you think This will reduce their individual sizes? We could also rely on YouTube video for now till we agree on the way to move forward.

**Comment by akenmorris** [comment]
Yes, that sounds good, but we will need to split the others as well as I see several GIFs that are almost 60MB in the repo.

**Comment by akenmorris** [comment]
This is a real pain, it looks like the limit is more like 5MB:

http://progsharing.blogspot.com/2018/06/gifs-on-github-pages-content-length.html

**Comment by sheryjoe** [comment]
Let's reduce the resolutions of these gifs and have a link to a high res youtube video. I think we should move to a static site generator hosted on SCI sometime soon to mitigate these content limitations.

**Comment by AtefehKashani** [comment]
In the Build/BUILD.md the link to "Adding to PATH" is not working I assume it to be in the "How to" section of the getting started but it is missed.  This was a part of Gettingstarted.md .

**Comment by sheryjoe** [comment]
> In the Build/BUILD.md the link to "Adding to PATH" is not working I assume it to be in the "How to" section of the getting started but it is missed. This was a part of Gettingstarted.md .

Fixed. Good catch!

**Comment by akenmorris** [comment]
> Let's reduce the resolutions of these gifs and have a link to a high res youtube video.

I don't have access to the original videos except for studio.

**Comment by sheryjoe** [comment]
@AtefehKashani @jadie1 could you upload the original videos for your gifs/mp4 to the below folder?

https://drive.google.com/drive/folders/101pBP085XNKIQe7HZw_gOvMvJTl47vxC?usp=sharing

**Comment by sheryjoe** [comment]
@akenmorris for any video on youtube, please add a brief description on what the video is show casing.

**Comment by akenmorris** [comment]
Sounds good.  Are you sure you want GIFs as well as these will not be obvious that you can click on them to load a video.  What they suggested in those threads was to put the video on youtube, then take a screenshot of the video with the red play button over it.  Then it is obvious to the user that it's a video they can click and watch.

**Comment by sheryjoe** [comment]
This is even better. The red button will entice the user to click and play! and without keeping gifs on our repo. I like this. Let's do it.

**Comment by akenmorris** [comment]
How's this as an example:

https://github.com/SCIInstitute/ShapeWorks/blob/org-doc/Documentation/GettingStarted.md

**Comment by sheryjoe** [comment]
Looks good!

**Pull request merged**